src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSourceBase.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // Base class for the three "data table sources" used by the extension:
+    //
+    // 1. The "real" source, which adds entries to the error list.
+    // 2. A "stub" data source whose sole purpose is to cause the Supression State
+    //    column to be displayed.
+    // 3. A "stub" data source whose sole purpose is to cause the Category column
+    //    to be displayed.
+    //
+    // For an explanation of why these three sources are necessary, see the comment
+    // near the top of SarifResultTableEntry.cs
+    internal abstract class SarifTableDataSourceBase : ITableDataSource
+    {
+        [Import]
+        private ITableManagerProvider TableManagerProvider { get; set; } = null;
+
+        [ImportMany]
+        private IEnumerable<ITableControlEventProcessorProvider> TableControlEventProcessorProviders { get; set; } = null;
```

Comment:
What are you using this for?

Reply:
Good catch. This was existing code that I pushed down from the concrete `SarifTableDataSource` class. The stub classes certainly don't need it -- and in fact I now see that this is a private imported property that is not used by the importing class. So I will remove it.

---
In reply to: [538011095](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538011095) [](ancestors = 538011095)